src/app/shared/components/template/services/template-action.service.ts

```diff
@@ -0,0 +1,229 @@
+        return this.setLocalVariable(key, value);
+      case "set_global":
+        console.log("[SET GLOBAL]", key, value);
+        return this.container.templateService.setGlobal(key, value);
+      case "go_to":
+        return this.container.templateNavService.handleNavAction(action, this.container);
+      case "pop_up":
+        return this.container.templateNavService.handlePopupAction(action, this.container);
+      case "set_field":
+        console.log("[SET FIELD]", key, value);
+        return this.container.templateService.setField(key, value);
+      case "set_theme":
+        return this.container.templateService.setTheme(
+          this.container.template,
+          "set_theme",
+          action.args
+        );
+      case "start_tour":
+        return this.container.tourService.startTour(key);
+      case "trigger_actions":
+        const triggeredActions: FlowTypes.TemplateRowAction[] = args[0] as any;
+        // add actions to end of existing action queue for processing after current queue complete
+        triggeredActions.forEach((a) =>
+          this.actionsQueue.push({ ...a, _triggeredBy: action._triggeredBy })
+        );
+        return;
+      case "emit":
+        const [emit_value, emit_from] = args;
+        let container: TemplateContainerComponent = this.container;
+        if (emit_from) {
+          // emit from the named template container instead of this one if specified (assumed sibling of current)
+          const targetContainer = container.children[emit_from];
+          if (targetContainer) {
+            action.args = [emit_value];
+            container = targetContainer;
+          }
+        }
+        let { parent, row, name, template, templatename } = container;
+        console.log("[EMIT]", `${name || templatename}:${emit_value}`);
+        if (emit_value === "completed") {
+          // write completions to the database for data tracking
+          await this.container.templateService.recordEvent(template, "emit", emit_value);
+        }
+        if (parent) {
+          // continue to emit any actions to parent where defined
+          log(
+            "Emiting",
+            emit_value,
+            ` from ${row?.name || "(no row)"} to parent ${parent?.name || "(no parent)"}`,
+            parent
+          );
+          if (row && row.action_list) {
+            const actionsForEmittedEvent = row.action_list.filter((a) => a.trigger === emit_value);
+            log("Excuting actions matching event ", { emit_value, actionsForEmittedEvent });
+            // process in parallel, do not return/await
+
+            // TODO - check if should be parent row or current row
+            parent.templateActionService.handleActions(actionsForEmittedEvent, row);
+          } else {
+            log("No action list for row ", row, "on template name ", name);
+            // NOTE - CC 2021-08-05 - Remove default action passing to parent
+            // parent.handleActions([action], row);
```

Comment:
This is the main change made to parent action handling (commenting out default behaviour to pass to parent).

I'm not sure exactly why we had it before, so possible to have knock-on effects